sdks/java/core/src/main/java/org/apache/beam/sdk/runners/inprocess/InProcessSideInputContainer.java

```diff
@@ -173,41 +176,43 @@ private void updatePCollectionViewWindowValues(
     }
   }
 
-  private final class SideInputContainerSideInputReader implements SideInputReader {
+  private final class SideInputContainerSideInputReader implements ReadyCheckingSideInputReader {
     private final Collection<PCollectionView<?>> readerViews;
 
     private SideInputContainerSideInputReader(Collection<PCollectionView<?>> readerViews) {
       this.readerViews = ImmutableSet.copyOf(readerViews);
     }
 
     @Override
+    public boolean allViewsReadyInWindow(final BoundedWindow elementWindow) {
+      for (PCollectionView<?> view : readerViews) {
+        try {
+          BoundedWindow viewWindow =
+              view.getWindowingStrategyInternal().getWindowFn().getSideInputWindow(elementWindow);
```

Comment:
`isReady(view, sideInputWindow)` is symmetric with `get(view, sideInputWindow)`. Changed to that.